src/platform/Linux/OTAImageProcessorImpl.cpp

```diff
@@ -82,6 +84,35 @@ CHIP_ERROR OTAImageProcessorImpl::ProcessBlock(ByteSpan & block)
     return CHIP_NO_ERROR;
 }
 
+bool OTAImageProcessorImpl::IsFirstImageRun()
+{
+    OTARequestorInterface * requestor = chip::GetRequestorInstance();
+    if (requestor == nullptr)
+    {
+        return false;
+    }
+
+    return requestor->GetCurrentUpdateState() == OTARequestorInterface::OTAUpdateStateEnum::kApplying;
+}
+
+CHIP_ERROR OTAImageProcessorImpl::ConfirmCurrentImage()
+{
+    OTARequestorInterface * requestor = chip::GetRequestorInstance();
+    if (requestor == nullptr)
+    {
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    uint32_t currentVersion;
+    ReturnErrorOnFailure(DeviceLayer::ConfigurationMgr().GetSoftwareVersion(currentVersion));
+    if (currentVersion != requestor->GetTargetVersion())
+    {
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    return CHIP_NO_ERROR;
```

Comment:
I wonder if this function should copy `kImageExecPath` image to the initial image location to approve the update and make further restarts not "rollback" to the old image. But maybe that could be a followup.

Reply:
So I had thought about some of this. One things is getting the location of the executable wasn't so straight forward (usually folks are probably calling from top-level dir and executable is somewhere under out/ so calling `getcwd` only got you to the top-level location). Figuring this out required more time than I had wanted to spend on this :) 

Also, the rolling back vs not.. if we did copy to the initial image location, that means if you want to run OTA again after terminating the app, you'd have to build a newer version otherwise the next image will never pass Confirm. I couldn't quite decide how we should handle this situation either so just left it as this second location. I can definitely file an issue to track this for later.

Reply:
One more thought on this is if confirming the new image fails, we might want to consider rolling back to the previous image. I guess this should all be captured by a followup.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/16516